<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8" />
<meta name="google-site-verification" content="CAJSXz1q9IoKgvnzjtqfn7KaJhAJMC9sCywfxWYum7s" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<title>Python Common Built-in Exceptions</title>
<meta name="description" content="In Python, exceptions are raised when an error or unexpected situation occurs during the execution of a program. Here are some common built-in exceptions in Python:">
<meta name="robots" content="index, follow, max-image-preview:large, max-snippet:-1, max-video-preview:-1" />
<link rel="canonical" href="https://www.devbabu.com/shorts/python/python-common-built-in-exceptions.html">
<link rel="icon" href="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/favicon.png" sizes="32x32" />
<link rel="icon" href="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/favicon.png" sizes="192x192" />
<link rel="apple-touch-icon" href="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/favicon.png" />
<meta name="msapplication-TileImage" content="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/favicon.png" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/style.min.css" />
</head>
<body>
<header class="header">
    <div class="wrap d-flex">
        <div class="logo">
            <a href="https://www.devbabu.com">DevBabu.Com</a>
        </div>
        <div class="right-action d-flex">
            <div class="search">
                <form action="/" method="GET" class="d-flex">
                    <input class="s-input" type="text" name="s" placeholder="Search..." autocomplete="off" required />
                    <button class="s-btn" type="submit">
                        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
                            <path fill-rule="evenodd" d="M9 3.5a5.5 5.5 0 1 0 0 11 5.5 5.5 0 0 0 0-11ZM2 9a7 7 0 1 1 12.452 4.391l3.328 3.329a.75.75 0 1 1-1.06 1.06l-3.329-3.328A7 7 0 0 1 2 9Z" clip-rule="evenodd" />
                        </svg>
                    </button>
                </form>
            </div>
            <div class="theme">
                <button class="theme-btn">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="theme-moon">
                        <path fill-rule="evenodd" d="M7.455 2.004a.75.75 0 0 1 .26.77 7 7 0 0 0 9.958 7.967.75.75 0 0 1 1.067.853A8.5 8.5 0 1 1 6.647 1.921a.75.75 0 0 1 .808.083Z" clip-rule="evenodd" />
                    </svg>
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="theme-sun">
                        <path d="M10 2a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 10 2ZM10 15a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 10 15ZM10 7a3 3 0 1 0 0 6 3 3 0 0 0 0-6ZM15.657 5.404a.75.75 0 1 0-1.06-1.06l-1.061 1.06a.75.75 0 0 0 1.06 1.06l1.06-1.06ZM6.464 14.596a.75.75 0 1 0-1.06-1.06l-1.06 1.06a.75.75 0 0 0 1.06 1.06l1.06-1.06ZM18 10a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 18 10ZM5 10a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 5 10ZM14.596 15.657a.75.75 0 0 0 1.06-1.06l-1.06-1.061a.75.75 0 1 0-1.06 1.06l1.06 1.06ZM5.404 6.464a.75.75 0 0 0 1.06-1.06l-1.06-1.06a.75.75 0 1 0-1.061 1.06l1.06 1.06Z" />
                    </svg>
                </button>
            </div>
        </div>
    </div>
</header>
<div class="container">
<nav class="navigation">
<ul>
<li><a href="/">Home</a></li>
<li><a href="/shorts/index.html">Shorts</a></li>
<li><a href="./index.html">PYTHON</a></li>
</ul>
</nav>
<article>
<h1>Python Common Built-in Exceptions</h1><p>In Python, exceptions are raised when an error or unexpected situation occurs during the execution of a program. <strong>Each type of exception is represented by a specific class, and these classes are organized in a hierarchy</strong>. Here are <strong>some common built-in exceptions in Python</strong>:</p>

<ol>
<li><p><strong><code>SyntaxError</code>:</strong></p>

<ul>
<li>Raised when there is a syntax error in the code.</li>
</ul>

<pre class="language-python"><code>print("Hello, World"  # Missing closing parenthesis
</code></pre></li>
<li><p><strong><code>IndentationError</code>:</strong></p>

<ul>
<li>Raised when there is an indentation error in the code.</li>
</ul>

<pre class="language-python"><code>def example():
print("Indented incorrectly")
</code></pre></li>
<li><p><strong><code>NameError</code>:</strong></p>

<ul>
<li>Raised when a local or global name is not found.</li>
</ul>

<pre class="language-python"><code>print(undefined_variable)
</code></pre></li>
<li><p><strong><code>TypeError</code>:</strong></p>

<ul>
<li>Raised when an operation or function is applied to an object of an inappropriate type.</li>
</ul>

<pre class="language-python"><code>result = "10" + 5
</code></pre></li>
<li><p><strong><code>ZeroDivisionError</code>:</strong></p>

<ul>
<li>Raised when division or modulo by zero occurs.</li>
</ul>

<pre class="language-python"><code>result = 10 / 0
</code></pre></li>
<li><p><strong><code>ValueError</code>:</strong></p>

<ul>
<li>Raised when a built-in operation or function receives an argument of the correct type but an inappropriate value.</li>
</ul>

<pre class="language-python"><code>number = int("abc")
</code></pre></li>
<li><p><strong><code>IndexError</code>:</strong></p>

<ul>
<li>Raised when a sequence subscript is out of range.</li>
</ul>

<pre class="language-python"><code>my_list = [1, 2, 3]
element = my_list[5]
</code></pre></li>
<li><p><strong><code>FileNotFoundError</code>:</strong></p>

<ul>
<li>Raised when an attempt is made to open a file or directory that cannot be found.</li>
</ul>

<pre class="language-python"><code>with open("nonexistent_file.txt", "r") as file:
   content = file.read()
</code></pre></li>
<li><p><strong><code>KeyError</code>:</strong></p>

<ul>
<li>Raised when a dictionary key is not found.</li>
</ul>

<pre class="language-python"><code>my_dict = {"name": "John"}
value = my_dict["age"]
</code></pre></li>
<li><p><strong><code>AttributeError</code>:</strong></p>

<ul>
<li>Raised when an attribute reference or assignment fails.</li>
</ul>

<pre class="language-python"><code>value = "Hello, World"
length = value.length  # AttributeError: 'str' object has no attribute 'length'
</code></pre></li>
</ol>

<p>These are just a few examples of built-in exceptions in Python. Understanding the types of exceptions and their meanings helps in debugging and handling errors more effectively in your Python programs. You can catch and handle exceptions using <code>try</code>, <code>except</code>, <code>else</code>, and <code>finally</code> blocks.</p>
</article>
</div>
<footer>
    <ul class="footer-nav d-flex">
        <li><a href="/about-us/">About</a></li>
        <li><a href="/privacy-policy/">Privacy</a></li>
        <li><a href="/contact-us/">Contact</a></li>
    </ul>
    <div class="extra" id="extraEl"></div>
</footer>
<script src="https://cdn.jsdelivr.net/gh/chandan-tudu/shorts@latest/assets/script.min.js"></script>
</body>
</html>